src/Filter/FilterInterface.php

```diff
@@ -24,10 +25,7 @@ interface FilterInterface
 
     public const CONDITION_AND = 'AND';
 
-    /**
-     * @phpstan-param array{type?: int|null, value?: mixed} $filterData
```

Comment:
Just realised that `type` could not exists, but not sure if that makes sense. 

With the current implementation `FilterData::getType()` would return `null` if `type` is not present in the array.

Reply:
null and no value should behave the same IMHO.